core-web/libs/sdk/analytics/src/lib/shared/dot-content-analytics.http.ts

```diff
@@ -8,23 +8,24 @@ import { DotAnalyticsConfig, PageViewEvent } from './analytics.model';
  * @returns A promise that resolves to the response from the server
  */
 export const sendAnalyticsEventToServer = async (
-    data: PageViewEvent,
-    options: DotAnalyticsConfig
+    data: Record<string, unknown>,
+    options: DotContentAnalyticsConfig
 ): Promise<Response> => {
-    const eventData = {
+    const serverEvent: ServerEvent = {
         ...data,
-        timestamp: new Date().toISOString()
+        timestamp: new Date().toISOString(),
+        key: options.apiKey
     };
 
     if (options.debug) {
-        console.warn('DotAnalytics: Event sent:', eventData);
+        console.warn('DotAnalytics: Event sent:', serverEvent);
     }
 
     try {
         return await fetch(`${options.server}${ANALYTICS_ENDPOINT}`, {
```

Comment:
Add validation to the status, example:

try {

    const response = await fetch(`${options.server}${ANALYTICS_ENDPOINT}`, {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify(serverEvent)
        });

    if (response.ok) {
        return response;
    } else {
        throw new Error(response.status);
    }
} catch (error) {
        console.error('DotAnalytics: Error sending event:', error);
        throw error;
    }

Reply:
Added.